src/main/java/frc/robot/subsystems/Shooter.java

```diff
@@ -0,0 +1,214 @@
+    rightConfig.Slot0.kP = prefShooter.rightShooterP.getValue();
+    rightConfig.Slot0.kI = prefShooter.rightShooterI.getValue();
+    rightConfig.Slot0.kD = prefShooter.rightShooterD.getValue();
+
+    rightConfig.MotionMagic.MotionMagicAcceleration = 400;
+    rightConfig.MotionMagic.MotionMagicJerk = 4000;
+
+    pivotConfig.Slot0.kP = prefShooter.leftShooterP.getValue();
+    pivotConfig.Slot0.kI = prefShooter.leftShooterI.getValue();
+    pivotConfig.Slot0.kD = prefShooter.leftShooterD.getValue();
+
+    pivotConfig.SoftwareLimitSwitch.ForwardSoftLimitEnable = true;
+    pivotConfig.SoftwareLimitSwitch.ForwardSoftLimitThreshold = constShooter.PIVOT_FORWARD_LIMIT.in(Units.Rotations);
+
+    pivotConfig.SoftwareLimitSwitch.ReverseSoftLimitEnable = true;
+    pivotConfig.SoftwareLimitSwitch.ReverseSoftLimitThreshold = constShooter.PIVOT_FORWARD_LIMIT.in(Units.Rotations);
+
+    leftMotor.getConfigurator().apply(leftConfig);
+    rightMotor.getConfigurator().apply(rightConfig);
+    pivotMotor.getConfigurator().apply(pivotConfig);
+
+    leftMotor.setInverted(leftInvert);
+    rightMotor.setInverted(rightInvert);
+  }
+
+  /**
+   * Sets both shooting motors to try to get to their previously-assigned desired
+   * speeds.
+   */
+  public void getUpToSpeed() {
+    if (desiredLeftVelocity.in(Units.RotationsPerSecond) == 0
+        && desiredRightVelocity.in(Units.RotationsPerSecond) == 0) {
+      setShootingNeutralOutput();
+    } else {
+      leftMotor.setControl(motionMagicRequest.withVelocity(desiredLeftVelocity.in(Units.RotationsPerSecond)));
+      rightMotor.setControl(motionMagicRequest.withVelocity(desiredRightVelocity.in(Units.RotationsPerSecond)));
+    }
+  }
+
+  public void setLeftShooterIntakeVoltage(double voltage) {
+    leftMotor.setControl(voltageRequest.withOutput(voltage));
+
+  }
+
+  public void setRightShooterIntakeVoltage(double voltage) {
+    rightMotor.setControl(voltageRequest.withOutput(voltage));
+  }
+
+  /**
+   * Sets all of the shooting motors to neutral.
+   */
+  public void setShootingNeutralOutput() {
+    leftMotor.setControl(new NeutralOut());
+    rightMotor.setControl(new NeutralOut());
+  }
+
+  /**
+   * @return The current velocity of the left shooter motor. <b> Units: </b>
+   *         Rotations per second
+   */
+  public double getLeftShooterVelocity() {
+    return leftMotor.getVelocity().getValueAsDouble();
+  }
+
+  /**
+   * @return The current velocity of the right shooter motor. <b> Units: </b>
+   *         Rotations per second
+   */
+  public double getRightShooterVelocity() {
+    return rightMotor.getVelocity().getValueAsDouble();
+  }
+
+  /**
+   * @return If the left shooter motor is at its desired velocity
+   */
+  public boolean isLeftShooterUpToSpeed() {
+    return (Math.abs(
+        getLeftShooterVelocity()
+            - desiredLeftVelocity.in(Units.RotationsPerSecond))) <= constShooter.UP_TO_SPEED_TOLERANCE
+                .in(Units.RotationsPerSecond);
```

Comment:
I believe in the case of comparisons, the unit conversion is done automatically so you can just compare the objects.